src/compression.rs

```diff
@@ -557,3 +586,251 @@ pub fn compress_graph<K: Kmer, D: Clone + Debug, S: CompressionSpec<D>>(
 ) -> DebruijnGraph<K, D> {
     CompressFromGraph::<K, D, S>::compress_graph(stranded, spec, old_graph, censor_nodes)
 }
+
+
+//////////////////////////////
+// Compress from Hash a new Struct
+//////////////////////////////
+/// Generate a compressed DeBruijn graph from hash_index
+struct CompressFromHash<'a, K: 'a + Kmer, D: 'a, S: CompressionSpec<D>> {
+    stranded: bool,
+    k: PhantomData<K>,
+    d: PhantomData<D>,
+    spec: S,
+    available_kmers: BitSet,
+    index: &'a boomphf::BoomHashMap2<K, Exts, D>,
+}
+
+/// Compression of paths in Debruijn graph
+impl<'a, K: Kmer, D: Clone + Debug, S: CompressionSpec<D>> CompressFromHash<'a, K, D, S> {
+    fn get_kmer_data(&'a self, kmer: &K) -> (&Exts, &'a D) {
+        match self.index.get_data_for_kmer(kmer) {
+            Some(data) => data,
+            None => panic!("couldn't find kmer {:?}", kmer),
+        }
+    }
+
+    fn get_kmer_id(&self, kmer: &K) -> Result<usize, usize> {
```

Comment:
does the Err case of this Result<usize,usize> have any meaning?  If not, this should just return an Option<usize>